File: c DSA/JumpSearch.cpp
//WAP TO USE JUMP SEARCH ALGO FOR SEARCHING AN ELEMENT X IN ARRAY
#include<stdio.h>
#include<math.h>
int  jumpSearch(int a[],int n)
{ int x,pointer=0;
	printf("enter ele you want to search\n");
	scanf("%d",&x);
	int step= sqrt(n);//ASSUMING STEP SIZE TO  BE SQUARE ROOT OF TOTAL ELEMENTS OF ARRAY
	
	printf("step size is %d \n",step);

	while(a[pointer]<=a[n-1])
	{ 
	if(a[pointer]<x&&a[pointer]<=a[n-1])
	{
		pointer=pointer+step;
		printf("pointer= %d \n",pointer);
	}
		
	if(a[pointer]==x)
	{
	
	printf("ele %d is found and present at %d index",x,pointer);
	return 0;
}
	if(a[pointer]>x&&a[pointer]>=0)
	{
		pointer=pointer-step;
		
		int upper=pointer+step;
		printf("upper= %d",upper);
		printf("pointer= %d \n",pointer);
		while(pointer<upper)
		{ if(a[pointer]==x) 
		{ printf("ele %d is found and present at %d index\n",x,pointer);
			return 0;
		}
		pointer++;
			printf("pointer = %d \n",pointer);
		
		}
		
	}
		
		
	}
	printf("%d is not found in array\n",x);
}
int main()
{
	int a[40],n,i;
	printf("enter no of ele in array\n");
	scanf("%d",&n);
	for(i=0;i<n;i++)
	{
		printf("enter %d index ele\n",i);
		scanf("%d",&a[i]);
	}
	jumpSearch(a,n);
}

File: python DSA/JumpSearch.py
# -*- coding: utf-8 -*-
"""
Created on Thu Dec  3 15:11:36 2020

@author: Venkateswarlu
"""
#WAP TO SEARCH AN ELEMENT BY JUMP SEARCH 
#TIME COMPLEXITY -- O(sqrt(n))
#SPACE COMPLEXITY IS O(1) AND ASSUMING  STEP AS SQUARE ROOT
import math
def jumpSearch( a,n):
    x=int(input("enter ele you want to search"))
    print(x)
    position=0
    step = (int) (math.sqrt(x))
    while(position<=(n-1)):
        if(a[position] < x):
            position=position+step
            print("postion= "+str(position))
            print(position)
        if(a[position]==x):
            print("ele "+str(x)+"is present at index= "+str(position))
            return 0
        if(a[position]>x):
            position=position-step
            print("postion= "+str(position))
            upper=position+step
            while( position<upper):
                if(a[position]==x):
                    print("ele "+str(x)+"is present at index="+str(position))
                    return 0
                position=position+1
    print("ele "+str(x)+"is not found in list")
            
            
            
    
    
n=int(input("enter no of ele in list\n"))
a= [ ]
for i in range(n):
    ele=(int)(input("enter ele of "+str(i)+"index "))
    a.append(ele)
jumpSearch(a,n)
    
